src/TriggerBinding/SqlTableChangeMonitor.cs

```diff
@@ -254,14 +254,35 @@ private async Task RunChangeConsumptionLoopAsync()
                     await connection.OpenAsync(token);
                     this._logger.LogDebugWithThreadId("END OpenChangeConsumptionConnection");
 
+                    bool forceReconnect = false;
                     // Check for cancellation request only after a cycle of checking and processing of changes completes.
                     while (!token.IsCancellationRequested)
                     {
+                        bool isConnected = await connection.TryEnsureConnected(forceReconnect, this._logger, "ChangeConsumptionConnection", token);
+                        if (!isConnected)
+                        {
+                            // If we couldn't reconnect then wait our delay and try again
+                            await Task.Delay(TimeSpan.FromMilliseconds(this._pollingIntervalInMs), token);
```

Comment:
If the SQL service is unresponsive due to heavy load, will trying to reconnect constantly be acceptable? Should we limit the number of retries or have an exponential backoff?

Reply:
Yeah, user configuration is likely something we'll want to add. I'll make an issue and we can add that as a follow up item.